mantidproject/mantid
# Create RROA v3 with unused polarization correction properties removed

**Description of work.**
This PR remove properties from ReflectometryReductionOneAuto are not actually required any more now that they are taken from the parameters file. These are `CRho`, `CAlpha`, `CAp` and `CPp`. A new version of the algorithm has been created to do this. The related input boxes have been removed from the ISIS Reflectometry GUI and the PA and PNR options removed from the drop-down box. The drop-down box has then been replaced by a simple checkbox to indicate whether polarization corrections should be applied or not.

**Report to:**  Jos at ISIS

**To test:**
To test the changes in the algorithm run the following script. This will run RROA v3 with polarization corrections and compares the output with RROA v2 with polarization correction set to parameter file. These should be the same.  
```
ws = Load("offspec36698")
IvsQ_binned_RROA_v3, IvsQ_RROA_v3, IvsLam_RROA_v3= ReflectometryReductionOneAuto(InputWorkspace = ws, SummationType = "SumInQ", ReductionType = "DivergentBeam", AnalysisMode = "MultiDetectorAnalysis",ThetaIn = 0.3, ProcessingInstructions = "390-415",  MomentumTransferStep = 0.03, PolarizationAnalysis = True)
IvsQ_binned_RROA_v2, IvsQ_RROA_v2, IvsLam_RROA_v2 = ReflectometryReductionOneAuto(InputWorkspace = ws, SummationType = "SumInQ", ReductionType = "DivergentBeam", AnalysisMode = "MultiDetectorAnalysis",ThetaIn = 0.3, ProcessingInstructions = "390-415",  MomentumTransferStep = 0.03, PolarizationAnalysis = "ParameterFile", Version = 2)
CompareWorkspaces(IvsQ_binned_RROA_v3, IvsQ_binned_RROA_v2)
CompareWorkspaces(IvsQ_RROA_v3, IvsQ_RROA_v2)
CompareWorkspaces(IvsLam_RROA_v3, IvsLam_RROA_v2)
```
In addition open the reflectometry GUI and go to the settings tab and ensure there is a polarisation correction checkbox. The checkbox should be enabled when instrument is set to offspec, polref and crisp and disabled when set to inter or surf. 
- On the Runs tab, set the instrument to offspec and enter run number 44956 and angle 0.4. 
- Untick the polarisation correction checkbox.
- Click Process back on the Runs tab. When processing has finished, click the plot-rows button. You should get the output below.
- Repeat with the polarisation correction checkbox ticked and the plot should update to the second image below.

![polcor_false](https://user-images.githubusercontent.com/12895056/61854027-2d125300-aeb5-11e9-9f1e-f805bc2e0e94.png)

![polcor_true](https://user-images.githubusercontent.com/12895056/61854033-300d4380-aeb5-11e9-8214-c04b9d41dcde.png)

Refs #23818.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
